### PR TITLE
CNNGNS-14 - CNN-HEALTH Custom health check support

### PR DESCRIPTION
**Pull request by zoitec**
subsystem: CNN-HEALTH

Add option to run health check with custom check through the use of a custom class or a custom config object

Fixes: #6

Reviewed-By: Young, James <James.Young@turner.com>
Reviewed-By: AD Slaton <ad.slaton@turner.com>
Reviewed-By: Sean Joseph <sean.joseph@turner.com>

### CODE REVIEW

File: example/app.js [hunk 1]
```diff
@@ -14,6 +25,144 @@ function test() {
...
+                callback: function (json) {
+                    return json.version;
+                },
+                tick: function () {
+                    return fetch('http://www.cnn.com/_healthcheck')
+                        .then((response) => {
+                            let message;
+
+                            if (!response.ok) {
+                                message = util.format('BadResponse %s', response.status);
```
Comment (adslaton) [inline example/app.js]: No need to use util anymore. You can use a tick comment like so `BadResponse ${response.status}`

File: lib/plugins/checks/custom.check.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*global fetch*/
+'use strict';
+
+const status = require('./status'),
+    Check = require('./check'),
+    util = require('util');
```
Comment (adslaton) [inline lib/plugins/checks/custom.check.js]: Unused.

File: example/lib/customcheck/custom.check.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    tick () {
+        return new Promise((fulfill, reject) =>{
+                fetch('http://www.cnn.com/_healthcheck')
+                .then((response) => {
+                    let message;
+
+                    if (!response.ok) {
+                        message = util.format('BadResponse %s', response.status);
```
Comment (adslaton) [inline example/lib/customcheck/custom.check.js]: Use the tick comment.

File: lib/plugins/checks/custom.check.js (hunk 2 above)
Comment (zoitec) [reply lib/plugins/checks/custom.check.js]: done

Comment (adslaton) [reply lib/plugins/checks/custom.check.js]: Remove that line and we should be good to go.

Comment (zoitec) [reply lib/plugins/checks/custom.check.js]: done

**Review by adslaton — APPROVED** [review]

**Pull request merged**